nixops_aws/resources/ec2_rds_dbinstance.py

```diff
@@ -272,39 +272,41 @@ def _wait_for_dbinstance(self, dbinstance, state="available"):
             }:
                 raise Exception(
                     "RDS database instance ‘{0}’ in an error state (state is ‘{1}’)".format(
-                        dbinstance.id, dbinstance.status
+                        dbinstance["DBInstanceIdentifier"],
+                        dbinstance["DBInstanceStatus"],
                     )
                 )
-            if dbinstance.status == state:
-                break
+            if dbinstance["DBInstanceStatus"] == state:
+                return dbinstance
             time.sleep(6)
 
     def _copy_dbinstance_attrs(
         self, dbinstance, rds_security_groups, vpc_security_groups
     ):
         with self.depl._db:
-            self.rds_dbinstance_id = dbinstance.id
-            self.rds_dbinstance_allocated_storage = int(dbinstance.allocated_storage)
-            self.rds_dbinstance_instance_class = dbinstance.instance_class
-            self.rds_dbinstance_master_username = dbinstance.master_username
-            self.rds_dbinstance_engine = dbinstance.engine
-            self.rds_dbinstance_multi_az = dbinstance.multi_az
-            if dbinstance.subnet_group:
-                self.subnet_group = dbinstance.subnet_group.name
-
-            self.rds_dbinstance_port = int(dbinstance.endpoint[1])
-            self.rds_dbinstance_endpoint = "%s:%d" % dbinstance.endpoint
+            self.rds_dbinstance_id = dbinstance["DBInstanceIdentifier"]
```

Comment:
I'd have abstracted the AWS strings into a String Enum `enum.Enum`, rather than multiplying hardcoded references to it, that way, it's easy to change them in all the relevant places. But it can be done in a further PR IMHO.